packages/core-plugin-api/src/apis/definitions/AnalyticsApi.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiRef, createApiRef } from '../system';
+import { AnalyticsDomainValue } from '../../analytics/types';
+
+/**
+ * Represents an event worth tracking in an analytics system that could inform
+ * how users of a Backstage instance are using its features.
+ *
+ * Note that attributes about the Backstage user or about the plugin tracking
+ * the event are inferred and captured separately on the Analytics Domain and
+ * do not need to be passed on the event itself.
+ */
+export type AnalyticsEvent = {
+  /**
+   * A string that identifies the event being tracked by the type of action the
+   * event represents. Examples include:
+   *
+   * - view
+   * - click
+   * - filter
+   * - search
+   * - hover
+   * - scroll
+   */
+  verb: string;
+
+  /**
+   * A string that uniquely identifies the object that the verb or action is
+   * being taken on. Examples include:
+   *
+   * - The path of the page viewed
+   * - The url of the link clicked
+   * - The value that was filtered by
+   * - The text that was searched for
+   */
+  noun: string;
+
+  /**
+   * An optional numeric value relevant to the event that could be aggregated
+   * by analytics tools. Examples include:
+   *
+   * - The index or position of the clicked element in an ordered list
+   * - The percentage of an element that has been scrolled through
+   * - The amount of time that has elapsed since a fixed point
+   */
+  value?: number;
+
+  /**
+   * Optional context with any additional dimensions or metrics that could be
+   * forwarded on to analytics systems.
+   */
+  context?: AnalyticsEventContext;
+};
+
+/**
+ * An analytics event combined with domain attributes.
+ */
+export type DomainDecoratedAnalyticsEvent = AnalyticsEvent & {
+  /**
+   * Domain metadata relating to where the event was captured and by whom. This
+   * could include information about the route, plugin, or component in which
+   * an event was captured.
+   */
+  domain: AnalyticsDomainValue;
```

Comment:
Wondering a bit about the existence of both `context` and `domain`, how come they're both needed? I'm thinking you could for example move the known domain fields into the context. Is the purpose of the context to be things that are configured more locally?

Reply:
Yeah, your intuition is correct.  As someone writing an implementation of this API (potentially even an App Integrator), there's a general need to be able to distinguish between 1) information that is narrowly scoped to the event itself (`context`, in the current implementation) , 2) information pertaining to where the event took place (`domain`, in the current implementation), and 3) information about the person who triggered the event (not yet implemented).  Various analytics systems may treat/capture this information differently.

Reply:
Id did think it was the other way around to begin with tbh, as in `context` being what you call `domain`, given that it's the context within which the event happens 😅. Do you think it would make sense to flip them around and then find an alternate name for the local one, perhaps just `labels` or smth?

Reply:
Makes sense.  I went with `attributes`.  🤷‍♂️